lib/rubocop/result_cache.rb

```diff
@@ -188,14 +194,24 @@ def rubocop_checksum
             .select { |path| File.file?(path) }
             .sort!
             .each do |path|
-              content = File.binread(path)
-              digest << Zlib.crc32(content).to_s # mtime not reliable
+              digest << digest(path)
             end
           digest << RuboCop::Version::STRING << RuboCop::AST::Version::STRING
           digest.hexdigest
         end
     end
 
+    def digest(path)
+      if path.end_with?(*DL_EXTENSIONS)
+        # shared libraries often contain timestamps of when
+        # they were compiled and other non-stable data
+        path
```

Comment:
Hum, actually when we hash the `.rb` files, we don't care where they are located. So using the path as a "digest" mean that the `BUNDLE_PATH` now has influence, that may also be a problem for some cases. Maybe we should just extract a small name, e.g. `Zlib.crc32(File.basename(path)).to_s`.

Reply:
Yeah good point.